File: server/src/models/Category.go
package models

import (
	"time"

	"github.com/google/uuid"
	"gorm.io/gorm"
)

type Category struct {
	gorm.Model

	Id        uuid.UUID `gorm:"type:char(36);primaryKey"`
	Name      string    `gorm:"type:varchar(128)"`
	Code      string    `gorm:"type:varchar(100)"`
	CreatedAt time.Time // Automatically managed by GORM for creation time
	UpdatedAt time.Time // Automatically managed by GORM for update time
}

func (category *Category) BeforeCreate(tx *gorm.DB) (err error) {
	category.Id = uuid.New()
	return nil
}

File: client/models/category.ts
import mongoose from "mongoose";

const Category = new mongoose.Schema(
  {
    name: {
      type: String,
      require: [true, "Category Name required"],
    },
    code: {
      type: String,
      require: [true, "Category Code required"],
      unique: true,
    },
    description: {
      type: String,
      require: [false],
    },
  },
  {
    timestamps: true,
  }
);

const Categories =
  mongoose.models.categories || mongoose.model("categories", Category);

export default Categories;
